From 11a8816bece6dc03484dda82d21d51eb5d180056 Mon Sep 17 00:00:00 2001
From: gbhakteX <gurudattax.bhakte@intel.com>
Date: Wed, 26 Sep 2012 19:33:00 +0530
Subject: Audio-MW: Not able to play AAC-LD files through music app.

BZ: 56637

Currently MDP Intel optimized AAC decoder suite doesn't have AAC-LD
decoding capability.
This fix provides capability to invoke Google AAC decoder for AAC-LD
decoding.

Category: Feature Differentiation
Domain: AUDIO_MEDIA-Audio Framework
Origin: Internal
Upstream-Candidate: no, proprietary

Change-Id: I8eba0ca17294c5809942c2006f60402bc1d021e3
Signed-off-by: gbhakteX <gurudattax.bhakte@intel.com>
Orig-Change-Id: I928fcfa56969df384aff326b18078703430981f3
---
 media/libstagefright/MPEG4Extractor.cpp |    5 ++++-
 1 files changed, 4 insertions(+), 1 deletions(-)

diff --git a/media/libstagefright/MPEG4Extractor.cpp b/media/libstagefright/MPEG4Extractor.cpp
index 1527774..0c7849b 100644
--- a/media/libstagefright/MPEG4Extractor.cpp
+++ b/media/libstagefright/MPEG4Extractor.cpp
@@ -38,6 +38,7 @@
 #include <media/stagefright/MediaSource.h>
 #include <media/stagefright/MetaData.h>
 #include <utils/String8.h>
+#include <media/openmax/OMX_Audio.h>
 #ifdef ENABLE_AV_ENHANCEMENTS
 #include <QCMediaDefs.h>
 #include "include/ExtendedUtils.h"
@@ -2441,10 +2442,12 @@ status_t MPEG4Extractor::updateAudioTrackInfoFromESDS_MPEG4Audio(
 
     if (objectType == 31) {  // AAC-ELD => additional 6 bits
         objectType = 32 + br.getBits(6);
+    }
 #ifdef USE_INTEL_MDP
+    if ((objectType == OMX_AUDIO_AACObjectLD) || (objectType >= 31)) {
         mLastTrack->meta->setCString(kKeyMIMEType, MEDIA_MIMETYPE_AUDIO_AAC_EXTENDED);
-#endif
     }
+#endif
     mLastTrack->meta->setInt32(kKeyAACProfile, objectType);
 
 #ifdef QCOM_DIRECTTRACK
-- 
1.7.4.1

